src/pydvl/value/semivalues.py

```diff
@@ -0,0 +1,236 @@
+    :param done: Stopping criterion.
+    :param progress: Whether to display progress bars for each job.
+    :param job_id: id to use for reporting progress.
+    :return: Object with the results.
+    """
+    n = len(u.data.indices)
+    result = ValuationResult.empty(
+        algorithm=f"semivalue-{str(sampler)}-{coefficient.__name__}",
+        indices=sampler.indices,
+    )
+
+    samples = takewhile(lambda _: not done(result), sampler)
+    pbar = tqdm(disable=not progress, position=job_id, total=100, unit="%")
+    for idx, s in samples:
+        pbar.n = 100 * done.completion()
+        pbar.refresh()
+        marginal = (
+            (u({idx}.union(s)) - u(s)) * coefficient(n, len(s)) * sampler.weight(s)
+        )
+        result.update(idx, marginal)
+
+    return result
+
+
+def semivalues(
+    sampler: PowersetSampler,
+    u: Utility,
+    coefficient: SVCoefficient,
+    done: StoppingCriterion,
+    *,
+    n_jobs: int = 1,
+    config: ParallelConfig = ParallelConfig(),
+    progress: bool = False,
+) -> ValuationResult:
+    """
+    Computes semi-values for a given utility function and subset sampler.
+
+    :param sampler: The subset sampler to use for utility computations.
+    :param u: Utility object with model, data, and scoring function.
+    :param coefficient: The semi-value coefficient
+    :param done: Stopping criterion.
+    :param n_jobs: Number of parallel jobs to use.
+    :param config: Object configuring parallel computation, with cluster
+        address, number of cpus, etc.
+    :param progress: Whether to display progress bars for each job.
+    :return: Object with the results.
+
+    """
+    map_reduce_job: MapReduceJob[PowersetSampler, ValuationResult] = MapReduceJob(
+        sampler,
+        map_func=_semivalues,
+        reduce_func=lambda results: reduce(operator.add, results),
+        map_kwargs=dict(u=u, coefficient=coefficient, done=done, progress=progress),
+        config=config,
+        n_jobs=n_jobs,
+    )
+    return map_reduce_job()
+
+
+def shapley_coefficient(n: int, k: int) -> float:
+    return float(1 / math.comb(n - 1, k) / n)
+
+
+def banzhaf_coefficient(n: int, k: int) -> float:
+    return float(1 / 2 ** (n - 1))
+
+
+def beta_coefficient(alpha: float, beta: float) -> SVCoefficient:
+    # Leave this here in case we want to avoid depending on scipy
+    # n = n or 1024
+    # @lru_cache(maxsize=int(n * (n - 1) / 2))
+    # def B(a: int, b: int) -> float:
+    #     """Γ(a) * Γ(b) / Γ(a+b)"""
+    #     ii = np.arange(1, b)
+    #     return (ii / (a + ii)).prod() / a
+
+    B = sp.special.beta
+    const = B(alpha, beta)
+
+    def beta_coefficient_w(n: int, k: int) -> float:
+        """Beta coefficient"""
+        j = k + 1
+        w = B(j + beta - 1, n - j + alpha) / const
+        # return math.comb(n - 1, j - 1) * w * n
+        return float(w)
+
+    return cast(SVCoefficient, beta_coefficient_w)
+
+
+class SemiValueMode(str, Enum):
+    Shapley = "shapley"
```

Comment:
Isn't this redundant here since we already have other shapley methods?

I understand the reasoning behind adding it here and the fact that it doesn't take effort to do it but I think this should be removed here and documented.

Reply:
As discussed in person, and for the record, the idea is to leave this duplicate functionality for a while, while we test it, deprecate the old one and merge everything.